opus/opus.go

```diff
@@ -99,7 +112,52 @@ func (d *decoder) WriteSample(in Sample) error {
 		return nil
 	}
 	d.successiveErrorCount = 0
-	return d.w.WriteSample(d.buf[:n])
+
+	returnData := d.buf[:n]
+	if channels < d.targetChannels {
+		returnData = d.monoToStereo(returnData)
+	} else if channels > d.targetChannels {
+		returnData = d.stereoToMono(returnData)
+	}
+
+	return d.w.WriteSample(returnData)
+}
+
+func (d *decoder) monoToStereo(in media.PCM16Sample) media.PCM16Sample {
```

Comment:
These can be non-struct general methods.

Reply:
Done